src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -249,95 +280,66 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
+
+                        const width = firstTdBounds.left - r1c1FragmentBounds.left;
+                        r0c1.style.marginLeft = `-${width}px`;
+                        r1c1.style.marginLeft = `-${width}px`;
+                    }
                 }
-
-                // Force second column containers width to match visible portion of table
-                const firstVisibleTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns + 1})`);
-                if (firstVisibleTd) {
-                    const r1c1FragmentBounds = r1c1.getBoundingClientRect();
-                    const firstTdBounds = firstVisibleTd.getBoundingClientRect();
-
-                    const width = firstTdBounds.left - r1c1FragmentBounds.left;
-                    r0c1.style.marginLeft = `-${width}px`;
-                    r0c1.style.marginRight = `${width}px`;
-                    r1c1.style.marginLeft = `-${width}px`;
-                    r1c1.style.marginRight = `${width}px`;
-                }
-
-                it++;
-            } while (
-                Math.abs(currentWidth - lastWidth) > WIDTH_EPSILON ||
-                it < MAX_WIDTH_ITERATIONS
-            )
+            }
         }
 
-        if (fixed_columns || fixed_rows) {
-            const r1c0CellWidths = Array.from(
-                r1c0.querySelectorAll('table.cell-table > tbody > tr:first-of-type > *')
-            ).map(c => c.getBoundingClientRect().width);
+        if (!cycle) {
+            const currentWidth = parseInt(currentTableWith, 10);
+            const nextWidth = parseInt(getComputedStyle(r1c1Table).width, 10);
 
-            const r1c1CellWidths = Array.from(
-                r1c1.querySelectorAll('table.cell-table > tbody > tr:first-of-type > *')
-            ).map(c => c.getBoundingClientRect().width);
-
-            Array.from<HTMLElement>(
-                r0c0.querySelectorAll('table.cell-table > tbody > tr:first-of-type > *')
-            ).forEach((c, i) => this.setCellWidth(c, r1c0CellWidths[i]));
-
-            Array.from<HTMLElement>(
-                r0c0.querySelectorAll('table.cell-table > tbody > tr:last-of-type > *')
-            ).forEach((c, i) => this.setCellWidth(c, r1c0CellWidths[i]));
-
-            Array.from<HTMLElement>(
-                r0c1.querySelectorAll('table.cell-table > tbody > tr:first-of-type > *')
-            ).forEach((c, i) => this.setCellWidth(c, r1c1CellWidths[i]));
-
-            Array.from<HTMLElement>(
-                r0c1.querySelectorAll('table.cell-table > tbody > tr:last-of-type > *')
-            ).forEach((c, i) => this.setCellWidth(c, r1c1CellWidths[i]));
+            // If the table was resized and isn't in a cycle, re-run `handleResize`.
+            // If the final size is the same as the starting size from the previous iteration, do not
+            // resize the main table, instead just use as is, otherwise it will oscillate.
+            if (nextWidth !== currentWidth) {
```

Comment:
Tested these for scenarios involving decimal places and is seems to behave fine on all important targets, decimals or not.